### ISSUE TITLE
About the "does not match the expect variable shape ([0, ny, nx])" error

### ISSUE THREAD

**Issue by jypeter** [PCMDI/cmor#20]
@doutriaux1 , could you run my test script at the end of this mail and determine why I'm getting the following **does not match the expect variable shape ([0, 10, 18])** error?
```
  File "/usr/local/install/cdat/versions/cdat_install_uv-2.1.0_x86_64_gcc4_VB_13/lib/python2.7/site-packages/cmor/pywrapper.py", line 630, in write
    raise Exception,"error your data shape (%s) does not match the expect variable shape (%s)" % (str(osh),str(ogoodshape))
Exception: error your data shape ((1, 10, 18)) does not match the expect variable shape ([0, 10, 18])
```

I have had this error several times, always when trying to do something with the time axis, so *I guess the error is time axis related*, but I could be wrong

1. It's strange to get a **[0, ny, nx]** shape. I wonder if this could be a side effect of #16 

2. **This problem is very hard to debug** for the end user. I have used the following script to easily reproduce the error and it is clear that I am not passing any information about time values/bounds to cmor.write and that cmor.write should complain about it. But the error message should be more specific. And when I first encountered this problem, in another case, I had correctly specificied values for time! So I had to resort to trying to specify the time values in lots of different ways, till I eventually got rid of the error (for no obvious reason)

3. The error message (if you keep it) has a typo and should be "does not match the expect**ed** variable shape" :-)

**Note**: The script will work correctly if you uncomment the following lines. You can use it as an example, if need be, since it shows how udunits will automatically convert degF to K, and cmor2 will automatically update the axes to match what is requested in the CMOR tables
```
#               time_vals=numpy.array([15. + 30 * month]),
#               time_bnds=numpy.array([0. + 30 * month, 30. + 30 * month])
```

**Example script**

```
#!/usr/bin/env python

import numpy, cdms2, cmor

# Generate a dummy data matrix with '32 degF' everywhere except 2
# latitudinal bands around the equator
raw_data = numpy.ma.ones((1, 10, 18), numpy.float32) * 32
raw_data[:, 4:6, :] = numpy.ma.masked

# Generate axes for the data matrix
lat_axis = cdms2.createUniformLatitudeAxis(90, 10, -20)
lon_axis = cdms2.createUniformLongitudeAxis(-180, 18, 20)

cmor.setup(inpath='/home/users/jypeter/PMIP3_CMIP5/CMOR/cmip5-cmor-tables/Tables/',
           netcdf_file_action=cmor.CMOR_REPLACE)

cmor.dataset(experiment_id='lgm',
             institution='Institut Pierre Simon Laplace',
             institute_id='IPSL',
             model_id='ICE-MAKER-1',
             source='ICE-MAKER-1 atmosphere: JYP-007; ice: AMORINO-1',
             calendar='360_day',
             outpath='.',
             realization=1,
             forcing='N/A',
             parent_experiment_id='N/A', parent_experiment_rip='N/A',
             branch_time=0.,
             contact='JYP@LSCE',
             comment='This is some really dummy data',
             references='100 ways to freeze your ocean, The Ice Cream Journal')

cmor.load_table('CMIP5_Amon')

time_id = cmor.axis(table_entry= 'time',
                    units= 'days since 2000-01-01 00:00:00')

lat_id = cmor.axis(table_entry= 'latitude',
                   units= 'degrees_north',
                   coord_vals=lat_axis[:],
                   cell_bounds=lat_axis.getBounds())
lon_id = cmor.axis(table_entry= 'longitude',
                   units= 'degrees_east',
                   coord_vals=lon_axis[:],
                   cell_bounds=lon_axis.getBounds())

data_id = cmor.variable(table_entry='ts',
                        units='degF',
                        axis_ids=[time_id, lat_id, lon_id],
                        missing_value=raw_data.fill_value,
                        original_name='temp',
                        comment='A dummy variable from a dummy dataset')

# Write 3 time steps to the file (January to March). Add 18 degF to
# each time step (which should be converted to 10 K in the output
# file)
for month in range(3):
    cmor.write(var_id=data_id,
               data=raw_data + month * 18,
# Uncomment the 2 lines below in order to have a working script
#               time_vals=numpy.array([15. + 30 * month]),
#               time_bnds=numpy.array([0. + 30 * month, 30. + 30 * month])
)

file_path = cmor.close(data_id, file_name=True)

print '\nThe dummy data was successfully saved to', file_path

# The end
```

**Comment by dnadeau4** [comment]
@jypeter I solved this issue in branch "issues_20" #81.  I used your example to make a test.  The pywrapper message is now more explicit if time_values and time_bounds are missing when calling cmor_write.  I also encountered that difficulty before.  The message makes the developer look at their data, and do not realize that time_values, and time_bounds are missing when calling cmor_write().   I also translated your test for CMIP6 and CMOR3. :smile:
